site/content/docs/main/config/access-logging.md

```diff
@@ -0,0 +1,125 @@
+```
+
+After restarting Contour and successful validation, the new format will take effect in a short while.
+
+### Structured JSON Logging
+
+Contour allows you to choose from a set of JSON fields that will be expanded into Envoy templates and sent to Envoy.
+There is a default set of fields if you enable JSON logging, and you may customize which fields you log.
+
+The list of available fields are discoverable in the following objects:
+- [JSONFields][2] are fields that have built in mappings to commonly used envoy operators.
+- [envoySimpleOperators][3] are the names of simple envoy operators that don't require arguments, they are case-insensitive when configured.
+- [envoyComplexOperators][4] are the names of complex envoy operators that require arguments.
+
+The default list of fields is available at [DefaultFields][5].
+
+#### Enabling the Feature
+
+To enable the feature you have two options:
+
+- Add `--accesslog-format=json` to your Contour startup line.
+- Add `accesslog-format: json` to your configuration file.
+
+Without any further customization, the [default fields][5] will be used.
+
+#### Customizing Logged Fields
+
+To customize the logged fields, add a `json-fields` list of strings to your configuration file.
+If the `json-fields` key is not specified, the [default fields][5] will be configured.
+
+To use a value from [JSONFields][2] or [envoySimpleOperators][3], simply include the name of the value in the list of strings.
+The JSONFields are case-sensitive, but envoySimpleOperators are not.
+
+To use [envoyComplexOperators][4] or to use alternative field names, specify strings as key/value pairs like `"fieldName=%OPERATOR(...)%"`.
+
+Unknown field names in non key/value fields will result in validation errors, as will unknown Envoy operators in key/value fields.
+Note that the `DYNAMIC_METADATA` and `FILTER_STATE` Envoy logging operators are not supported at this time due to the complexity of their validation.
+
+See the [example config file][6] to see this used in context.
+
+#### Sample configuration file
+
+Here is a sample config:
+
+```yaml
+accesslog-format: json
+json-fields:
+  - "@timestamp"
+  - "authority"
+  - "bytes_received"
+  - "bytes_sent"
+  - "customer_id=%REQ(X-CUSTOMER-ID)%"
+  - "downstream_local_address"
+  - "downstream_remote_address"
+  - "duration"
+  - "method"
+  - "path"
+  - "protocol"
+  - "request_id"
+  - "requested_server_name"
+  - "response_code"
+  - "response_flags"
+  - "uber_trace_id"
+  - "upstream_cluster"
+  - "upstream_host"
+  - "upstream_local_address"
+  - "upstream_service_time"
+  - "user_agent"
+  - "x_forwarded_for"
+```
+
+## Using Access Log Formatter Extensions
```

Comment:
As with my comment on #3694, I don't think that we should include this until it's included in the Envoy version that Contour supports.